### PR TITLE
Add checkbox for splitting arrays

### PR DESCRIPTION
**Pull request by hannah-bulmer**
Solves issue https://github.com/elasticio/jsonata-transform-component/issues/12

Selecting the checkbox leaves array values as-is:

image
image

While deselecting it makes the array contents individual values:

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +35,8 @@
     "eslint-config-xo-space": "^0.15.0",
     "eslint-plugin-json": "^1.2.0",
     "eslint-plugin-mocha": "^4.12.1",
-    "mocha": "^3.5.3"
+    "mocha": "^3.5.3",
+    "sinon": "^7.4.2"
```
Comment (jhorbulyk) [inline package.json]: We don't use the `^` in our package.json files.  Can you remove all the `^` characters for both the new and old pacakges?   Also, now that you use the `@elastic.io/component-commons-library`, then `@elastic.io/jsonata-moment` is likely redundant and can be removed as well.

File: lib/actions/transform.js [hunk 2]
```diff
@@ -12,36 +10,21 @@ const PASSTHROUGH_BODY_PROPERTY = 'elasticio';
...
-    if (PASSTHROUGH_BODY_PROPERTY in message.body) {
-      throw new Error(`${PASSTHROUGH_BODY_PROPERTY} property is reserved \
-            if you are using passthrough functionality`);
-    }
-
-    message.body.elasticio = {};
-    Object.assign(message.body.elasticio, message.passthrough);
-  }
-  return message;
+  return Promise.resolve(eioUtils.newMessageWithBody(result));
```
Comment (jhorbulyk) [inline lib/actions/transform.js]: I see that you are copying the `Promise.resolve()` stuff that was already here but it probably makes sense as part of cleaning up this component to remove all the `Promise.resolve()` stuff and just return what is in the parentheses.

**Comment by jhorbulyk** [comment]
Based on a previous discussion ofline, we will hold of merging this PR until we address the arrays & passthrough issue.

**Comment by hannah-bulmer** [comment]
Closing this for now while it is still blocked

**Pull request closed without merging**